lib/oli/delivery/paywall.ex

```diff
@@ -535,13 +564,25 @@ defmodule Oli.Delivery.Paywall do
 
   def create_or_update_discount(attrs) do
     case get_discount_by!(%{
-      section_id: attrs.section_id,
-      institution_id: attrs.institution_id
-    }) do
+           section_id: attrs.section_id,
+           institution_id: attrs.institution_id
+         }) do
       nil -> %Discount{}
       discount -> discount
     end
     |> Discount.changeset(attrs)
     |> Repo.insert_or_update()
   end
+
+  def list_last_payment_codes_generated(count) do
```

Comment:
Yes, perfect. I will move it under the "list_payments" function. 

How about renaming it to just "list_last_payment_codes"? I think it would be understood that the number passed as a parameter refers to the number of payment codes I want to list. 

I will also add a documentation comment to clarify what the function itself does.